### DISCUSSION TITLE
Sort your favorite albums by date added

### DISCUSSION THREAD

**Discussion by d1rt** [LMS-Community/slimserver#941] [General]
I really miss sorting by date added to TIDAL favorite albums. At the moment albums in LMS are sorted alphabetically. But I use the Albums section in Tidal as a backlog, so I need the list of albums to open in LMS sorted by the date the album was added.

**Comment by mherger** [comment]
If you enable the Online Music Library Integration plugin you'll find all your favorite albums in My Music/New Albums etc.

**Reply by d1rt** [reply]
Thanks for the reply, but this method doesn't work for me.

TIDAL favorite albums:
<img width="1632" alt="Снимок экрана 2023-11-17 в 13 35 03" src="https://github.com/Logitech/slimserver/assets/2081931/ee5127cf-5c7e-4901-b489-f5eeb4a81505">

---------------------

LMS + Online Music Library Integration plugin / Music - New

<img width="696" alt="Снимок экрана 2023-11-17 в 13 50 38" src="https://github.com/Logitech/slimserver/assets/2081931/818d3364-f255-4457-989d-f9ffc0cea0af">

This is wrong sorting

---------------------

LMS + Online Music Library Integration plugin / Music - Albums

<img width="327" alt="Снимок экрана 2023-11-17 в 13 51 14" src="https://github.com/Logitech/slimserver/assets/2081931/0ca11fd8-487f-41f1-81ca-1bbb536d1dd7">

There is no Date Added option

**Reply by mherger** [reply]
That's not the "New Albums" menu, is it?

**Reply by d1rt** [reply]
The thing is, I don't want it sorted by year. I need the same order as I added albums to Tidal. There can be albums from any year, I just need the exact order in which they are added.

**Reply by michaelherger** [reply]
I'm sorry, I edited my message(s)... I hate it when others do that 😂. I was referring to the "New Music" menu. That one should be sorted by exact timestamp, not the publishing year.

**Reply by d1rt** [reply]
Yes that is New Music. As you can see, the list is similar to the Tidal list, but not 1:1. 

The first album is Apocalyptica, then J57 and so on

**Reply by mherger** [reply]
Hmm... I'd have to double check the timestamps they return. Maybe it's dates only, but no time of day. Could that be the problem here?

**Reply by d1rt** [reply]
Are you suggesting that Tidal is being given the wrong timestamp? But UAPP (Android app) for example has sorting by date added and it works perfectly. So the Tidal is giving the correct data.

**Reply by michaelherger** [reply]
I'm not saying it's wrong or right. I'm saying that we're using whatever data is stored in their `added` field. It's epoch time seconds. If you're familiar with SQL you could open the `library.db` in any SQLite client and check out the `timestamp` column in the `tracks` table.

**Reply by d1rt** [reply]
@michaelherger thanks for advice, will try

**Comment by d1rt** [comment]
> If you're familiar with SQL you could open the `library.db` in any SQLite client and check out the `timestamp` column in the `tracks` table.

Apparently this is the date and time the album was added to the LMS DB, because there are stacks of records with the same timestamp, it's obviously not the date I added albums to Tidal favorites.

<img width="138" alt="Снимок экрана 2023-11-17 в 15 57 38" src="https://github.com/Logitech/slimserver/assets/2081931/1fa9a055-38da-441b-86f3-193767a829cd">

**Reply by mherger** [reply]
The relevant column is `timestamp`, not `added_time`.

**Reply by d1rt** [reply]
Yes, you a right. And I found out that the New Music section actually shows the albums in chronological order as I added them to my favorites. The problem so far is that for some reason many albums are missing. 

For example:
1-5 albums from Tidal are skipped, 6 added to LMS, 7-9 skipped, 10 added, and so on.

That's why I thought the list was wrong at first.

**Reply by michaelherger** [reply]
If you search the albums table aren't they in there at all? Did you enable regular scanning of your online library? That doesn't happen in real-time, but only every few hours, or whenever you run a scan.

**Reply by d1rt** [reply]
Yeah, I checked, those albums aren't in the database.

<img width="1638" alt="Снимок экрана 2023-11-17 в 20 52 53" src="https://github.com/Logitech/slimserver/assets/2081931/62c1d853-ced1-4e3b-a178-427dd52c7ef0">

![Снимок экрана 2023-11-17 в 20 55 46](https://github.com/Logitech/slimserver/assets/2081931/ac02c671-bc04-4955-9b2c-b8bfcaa1cb25)


I did a manual rescan a few times today and I see the same New Music list in LMS

As you can see in the screenshots many Tidal albums didn't make it into the LMS Library for whatever reason

**Reply by michaelherger** [reply]
Please enable debug logging for `plugin.tidal` (Settings/Server/Logging in Material skin), then re-run an online only scan, disable logging again, and provide me with the resulting `scanner.log.zip` and `library.db` (https://www.dropbox.com/request/T3RctyzGgNg0oFDubq6a).

**Reply by d1rt** [reply]
done

**Reply by michaelherger** [reply]
Ok, something's wrong: you got an error back during that scan. I then looked into the log files on MySqueezebox.com. And I see the error, but I also see that about 15 minutes later the same request succeeded. Did you run another scan around (I guess) 14:07 your time? If so, would you have that log, too?

**Reply by d1rt** [reply]
These scans were automatic. I don't have the log from 14:07, but I do have the next one -- 14:13.

Sent. Thank you for your help

**Reply by michaelherger** [reply]
Unfortunately logging for `plugin.tidal` wasn't enabled at this point any more. But I got some information out of the log file anyway. You wouldn't have more than 2500 albums favorited in TIDAL, would you?... It looks as albums only through to `F` got imported. The last entry I can find is for an album called "Floreana". And if I look at your screenshots you can see that all missing albums are letters after `L`, and the important ones are before that.

My suspicion unfortunately is confirmed by looking at your `library.db`: there are 2507 albums, and ordered by title I'd see again albums up to "Floreana".

Here's the problem: import currently is limited to 2500 albums. I actually fear that the errors you sometimes get are due to requesting too much information from them. The processing might time out at some point.

**Reply by d1rt** [reply]
Thank you very much for researching the problem. Is there any way to add an option here to organize albums (My Music - Albums - Sort By) by date added (timestamp coulumn I guess)?

<img width="299" alt="Снимок экрана 2023-11-18 в 18 03 56" src="https://github.com/Logitech/slimserver/assets/2081931/5559ed58-828f-407e-9f34-a0c4299b02ac">

**Reply by michaelherger** [reply]
As the albums are missing from the library, this wouldn't even help. Can you browse all albums in the TIDAL menu in LMS? Can you scroll down to one of the missing albums?

**Reply by d1rt** [reply]
No. In TIDAL plugin ﻿﻿I can only see albums for the letters A and B. 

In Online Library plugin -- from A to F.
